tests/Test.ADAL.NET.Unit/InstanceDiscoveryTests.cs

```diff
@@ -179,12 +182,130 @@ public void TestInstanceDiscovery_WhenMultipleSimultaneousCallsWithTheSameAuthor
             {
                 PreferredNetwork = "login.microsoftonline.com",
                 PreferredCache = preferredCache,
-                Aliases = new string[] { "login.microsoftonline.com", "login.windows.net", "sts.microsoft.com" }
+                Aliases = new string[] {"login.microsoftonline.com", "login.windows.net", "sts.microsoft.com"}
             });
             var orderedList = await TokenCache.GetOrderedAliases(givenHost, false, new CallState(Guid.NewGuid())).ConfigureAwait(false);
             CollectionAssert.AreEqual(
-                new string[] { preferredCache, givenHost, "login.microsoftonline.com", "login.windows.net", "sts.microsoft.com" },
+                new string[] {preferredCache, givenHost, "login.microsoftonline.com", "login.windows.net", "sts.microsoft.com"},
                 orderedList);
         }
+
+        private void AddMockInstanceDiscovery(string host)
+        {
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
+            {
+                Method = HttpMethod.Get,
+                Url = $"https://{host}/common/discovery/instance",
+                ResponseMessage = new HttpResponseMessage(HttpStatusCode.OK)
+                {
+                    Content = new StringContent(
+                        @"{
+                        ""tenant_discovery_endpoint"" : ""https://login.microsoftonline.com/v1/.well-known/openid-configuration"",
```

Comment:
it should be {host} instead of login.microsoftonline.com

Reply:
Thanks for the hint that triggers an investigation. As it turns out that, the server response tends to present the querying `host` as-is into the `tenant_discovery_endpoint`, such as [this one](https://login.microsoftonline.com/common/discovery/instance?api-version=1.1&authorization_endpoint=https://login.windows.net/adfs/oauth2/authorize) and [that one](https://login.microsoftonline.com/common/discovery/instance?api-version=1.1&authorization_endpoint=https://login.microsoftonline.us/adfs/oauth2/authorize), but [not always so](https://login.microsoftonline.com/common/discovery/instance?api-version=1.1&authorization_endpoint=https://sts.windows.net/adfs/oauth2/authorize). So, using `{host}` instead of `login.microsoftonline.com` is not necessarily always a pricise choice.

But I guess none of above really matters. From the client-side perspective, ADAL does not have an opinion what endpoint URL the server returns. Client would just follow it. And in this particular case, ADAL does not actually send out a tenant discovery request at all. So, it doesn't matter. We could go with a `login.placeholder.com` and our test cases would still pass.

Lastly, this test value is currently a verbatim string containing a json blob. It would be very difficult to get string interpolation right.

All in all, I will probably change the current hardcoded `login.microsoftonline.com` to `placeholder` and that's it.

Reply:
nah, keep it the same way then.